src/emr_cluster.go

```diff
@@ -263,6 +281,37 @@ func (ec EmrCluster) GetInstanceGroups() []*emr.InstanceGroupConfig {
 	return instanceGroups
 }
 
+// GetEbsConfiguration turns a EbsConfigurationRecord into an emr.EbsConfiguration
+func GetEbsConfiguration(c *EbsConfigurationRecord) *emr.EbsConfiguration {
+	configs := c.EbsBlockDeviceConfigs
+
+	var emrConfigsArr []*emr.EbsBlockDeviceConfig
+
+	if configs != nil && len(configs) > 0 {
+		emrConfigsArr = make([]*emr.EbsBlockDeviceConfig, len(configs))
+
+		for i, config := range configs {
+			emrVolumeSpec := emr.VolumeSpecification{
+				Iops:       aws.Int64(config.VolumeSpecification.Iops),
+				SizeInGB:   aws.Int64(config.VolumeSpecification.SizeInGB),
+				VolumeType: aws.String(config.VolumeSpecification.VolumeType),
```

Comment:
Default values do work, albeit you have to define them as float even though the avro type is long...

That doesn't really solve the issue though as now iops can be unspecified but we don't know whether it should be present as we'll get at least default value not matter what.

Tbh, I'm keen on not enforcing this constraint as nothing really seems satisfactory.